File: flutter_app/lib/models/medication_order_model.dart
class MedicationOrder {
  final String patientId;
  final String pharmacyId;
  final String medicineId;

  MedicationOrder({
    required this.patientId,
    required this.pharmacyId,
    required this.medicineId,
  });

  factory MedicationOrder.fromJson(Map<String, dynamic> json) {
    return MedicationOrder(
      patientId: json['patientId'],
      pharmacyId: json['pharmacyId'],
      medicineId: json['medicineId'],
    );
  }
}

File: server/models/medication.order.model.js
const mongoose = require("mongoose");

const medicationOrderSchema = new mongoose.Schema(
  {
    patientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Patient",
      required: true,
    },
    pharmacyId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Pharmacy",
      required: true,
    },
    prescriptionId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Prescription",
      required: false,
    },
    medicineId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Medicine",
      required: false, // This is not required since an order might be linked to a prescription
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("MedicationOrder", medicationOrderSchema);
